src/pill/pill.tsx

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { PillProps } from "./types";
+import { Label, Wrapper } from "./pill.style";
+
+export const Pill = ({
+    type,
+    colorType = "black",
+    children,
+    icon,
+    ...otherProps
+}: PillProps): JSX.Element => {
+    return (
+        <Wrapper
+            data-testid={otherProps["data-testid"] || "pill"}
+            $type={type}
+            $color={colorType}
+            {...otherProps}
+        >
+            {icon && <div data-testid={"icon"}>{icon}</div>}
```

Comment:
Think u misunderstood my comment. I meant to add the test id on the icon in the test instead. Not add it to the component